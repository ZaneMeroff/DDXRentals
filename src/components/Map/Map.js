import React from 'react';
import map from '../../data/neon-map.png';
import './Map.scss'

const Map = (props) => {
  return (
    <div className='map' name='cap-hill'>
      <img id='map-image' src={ map } alt='neon map of denver with neighborhoods outlined'/>
      <div id='cap-hill' className='active-hood' onMouseEnter={() => props.updateNeighborhoodInfo('Capitol Hill')}></div>
      <div onMouseEnter={() => props.updateNeighborhoodInfo('Lower Highlands')}>
        <div className='active-hood' id='lohi-1' name='lohi'></div>
        <div id='lohi-2'></div>
      </div>
      <div onMouseEnter={() => props.updateNeighborhoodInfo('Park Hill')}>
        <div className='active-hood' name='park-hill' id='park-hill-1'></div>
        <div className='active-hood' name='park-hill' id='park-hill-2'></div>
      </div>
      <div onMouseEnter={() => props.updateNeighborhoodInfo('River North')}>
        <div className='active-hood' name='rino' id='rino-1'></div>
        <div id='rino-2'></div>
        <div id='rino-3'></div>
        <div id='rino-4'></div>
        <div id='rino-5'></div>
      </div>
    </div>
  )
}

export default Map;
