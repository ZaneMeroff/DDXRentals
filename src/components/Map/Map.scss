#map-image {
  height: 100vh;
  width: auto;
  margin-top: -5vh;

}

.map {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  overflow: hidden;
  position: relative;
  left: -30vh;
}

#cap-hill {
  border: 2px pink solid;
  height: 19vh;
  width: 13vh;
  position: relative;
  z-index: 10;
  top: -33vh;
  left: 1vh;
  margin-right: 33vh;
}

.active-hood:hover {
  background-color: rgba(255, 81, 237, 0.2)
}

#lohi-1 {
  border: 2px pink;
  width: 13.5vh;
  height: 15vh;
  position: relative;
  z-index: 10;
  top: -80.5vh;
  left: -32vh;
  border-left: 100px solid transparent;
}

// #lohi-2 {
//     width: 0;
//     height: 0;
//     border-bottom: 90px solid #202020;
//     border-left: 90px solid transparent;
//     position: relative;
//     z-index: 1;
//     top: -92.5vh;
//     left: -27.5vh;
//   }

  #park-hill-1 {
    background-color: #202020;
    // border-right: 100px solid transparent;
    width: 13.5vh;
    height: 7vh;
    position: relative;
    z-index: 11;
    top: -116vh;
    left: 34.5vh;
  }

  #park-hill-2 {
    border: 2px pink;
    width: 22.5vh;
    height: 46vh;
    position: relative;
    z-index: 10;
    top: -122vh;
    left: 24.5vh;
  }

  #rino-1 {
    border: 2px pink;
    width: 34.5vh;
    height: 20vh;
    position: relative;
    z-index: 10;
    top: -157vh;
    left: -5.5vh;
  }

  // #rino-2 {
  //   width: 0;
  //   height: 0;
  //   border-bottom: 114px solid #202020;
  //   border-left: 120px solid transparent;
  //   position: relative;
  //   z-index: 1;
  //   top: -184vh;
  //   left: 1.5vh;
  // }

  #rino-3 {
    width: 9vh;
    height: 25vh;
    background-color: #202020;
    position: relative;
    z-index: 10;
    top: -171vh;
    left: 19.5vh;
  }

  #rino-4 {
    width: 0;
    height: 0;
    border-bottom: 90px solid #202020;
    border-right: 75px solid transparent;
    position: relative;
    z-index: 10;
    top: -190vh;
    left: -5.5vh;
  }

  #rino-5 {
   width: 0;
   height: 0;
   border-top: 80px solid #202020;
   border-right: 33px solid transparent;
   position: relative;
   z-index: 10;
   top: -215vh;
   left: -6.5vh;
 }
