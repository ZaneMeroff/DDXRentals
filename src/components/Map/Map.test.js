import React from 'react';
import ReactDOM from 'react-dom';
import Map from './Map';
import { shallow } from 'enzyme';

describe('Map', () => {

  let mockUpdateNeighborhoodInfo;

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Map />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should be an instance of Map component', () => {
    mockUpdateNeighborhoodInfo = jest.fn()
    const wrapper = shallow(<Map
      updateNeighborhoodInfo={mockUpdateNeighborhoodInfo}
      />);
      expect(wrapper).toMatchSnapshot();
  });

});
