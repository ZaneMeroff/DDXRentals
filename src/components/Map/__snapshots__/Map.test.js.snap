// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map should be an instance of Map component 1`] = `
<div
  className="map"
  name="cap-hill"
>
  <img
    alt="neon map of denver with neighborhoods outlined"
    id="map-image"
    src="neon-map.png"
  />
  <div
    className="active-hood"
    id="cap-hill"
    onMouseEnter={[Function]}
  />
  <div
    onMouseEnter={[Function]}
  >
    <div
      className="active-hood"
      id="lohi-1"
      name="lohi"
    />
    <div
      id="lohi-2"
    />
  </div>
  <div
    onMouseEnter={[Function]}
  >
    <div
      className="active-hood"
      id="park-hill-1"
      name="park-hill"
    />
    <div
      className="active-hood"
      id="park-hill-2"
      name="park-hill"
    />
  </div>
  <div
    onMouseEnter={[Function]}
  >
    <div
      className="active-hood"
      id="rino-1"
      name="rino"
    />
    <div
      id="rino-2"
    />
    <div
      id="rino-3"
    />
    <div
      id="rino-4"
    />
    <div
      id="rino-5"
    />
  </div>
</div>
`;
